test/sequential/test-runner-run-inspect.js

```diff
@@ -0,0 +1,214 @@
+    execArgv: [`--inspect=127.0.0.1:${port}`],
+    expectedPort: port + 1
+  });
+
+  if (common.hasIPv6) {
+    port = debuggerPort + offset++ * 5;
+
+    await spawnRunner({
+      execArgv: [`--inspect=[::]:${port}`],
+      expectedPort: port + 1
+    });
+
+    port = debuggerPort + offset++ * 5;
+
+    await spawnRunner({
+      execArgv: [`--inspect=[::1]:${port}`],
+      expectedPort: port + 1
+    });
+  }
+
+  // These tests check that setting inspectPort in run
+  // would take effect and override port incrementing behavior
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: port + 2,
+    expectedPort: port + 2,
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'addTwo',
+    expectedPort: port + 2,
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'string',
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'null',
+    expectedPort: port + 1,
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'bignumber',
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'negativenumber',
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  await spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'bignumberfunc'
+  });
+
+  port = debuggerPort + offset++ * 5;
+
+  spawnRunner({
+    execArgv: [`--inspect=${port}`],
+    inspectPort: 'strfunc',
+  });
+}
+
+function runTest() {
+  const { run } = require('node:test');
+  let inspectPort = 'inspectPort' in process.env ? Number(process.env.inspectPort) : undefined;
+  let expectedError;
+
+  if (process.env.inspectPort === 'addTwo') {
+    inspectPort = common.mustCall(() => { return process.debugPort += 2; });
+  } else if (process.env.inspectPort === 'string') {
+    inspectPort = 'string';
+    expectedError = { name: 'RangeError', code: 'ERR_SOCKET_BAD_PORT' };
+  } else if (process.env.inspectPort === 'null') {
+    inspectPort = null;
+  } else if (process.env.inspectPort === 'bignumber') {
+    inspectPort = 1293812;
+    expectedError = { name: 'RangeError', code: 'ERR_SOCKET_BAD_PORT' };
+  } else if (process.env.inspectPort === 'negativenumber') {
+    inspectPort = -9776;
+    expectedError = { name: 'RangeError', code: 'ERR_SOCKET_BAD_PORT' };
+  } else if (process.env.inspectPort === 'bignumberfunc') {
+    inspectPort = common.mustCall(() => 123121);
+    expectedError = { name: 'RangeError', code: 'ERR_SOCKET_BAD_PORT' };
+  } else if (process.env.inspectPort === 'strfunc') {
+    inspectPort = common.mustCall(() => 'invalidPort');
+    expectedError = { name: 'RangeError', code: 'ERR_SOCKET_BAD_PORT' };
+  }
```

Comment:
Why use separate processes here? Can't we make multiple calls to `test.run` in the same process? Also there's a lot of duplicate for `expectedError`.

Reply:
we need to pass --inspect/--inspect-port etc as an arg to the process calling `test.run`